microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/integration/FailSafeCDICustomConfigSourceHealthTest.java

```diff
@@ -0,0 +1,92 @@
+import org.jboss.as.arquillian.container.ManagementClient;
+import org.jboss.eap.qe.microprofile.tooling.server.ModuleUtil;
+import org.jboss.eap.qe.microprofile.tooling.server.configuration.creaper.ManagementClientProvider;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.runner.RunWith;
+import org.wildfly.extras.creaper.core.online.OnlineManagementClient;
+
+@RunWith(Arquillian.class)
+@RunAsClient
+@ServerSetup({ MicroProfileFTSetupTask.class, FailSafeCDICustomConfigSourceHealthTest.SetupTask.class })
+public class FailSafeCDICustomConfigSourceHealthTest extends FailSafeCDIHealthDynamicBaseTest {
+    private static final String PROPERTY_FILENAME = "health.properties";
+    Path propertyFile = Paths.get(FailSafeCDICustomConfigSourceHealthTest.class.getResource(PROPERTY_FILENAME).getPath());
+    private byte[] bytes;
+
+    @Deployment(testable = false)
+    public static WebArchive createDeployment() {
+        WebArchive webArchive = ShrinkWrap
+                .create(WebArchive.class, FailSafeCDICustomConfigSourceHealthTest.class.getSimpleName() + ".war")
+                .addClasses(FailSafeDummyService.class, CDIBasedLivenessHealthCheck.class, CDIBasedReadinessHealthCheck.class)
+                .addAsWebInfResource(EmptyAsset.INSTANCE, "beans.xml");
+        return webArchive;
+    }
+
+    @Before
+    public void backup() throws IOException {
+        bytes = Files.readAllBytes(propertyFile);
+    }
+
+    @After
+    public void restore() throws IOException {
+        Files.write(propertyFile, bytes);
+    }
+
+    @Override
+    protected void setConfigProperties(boolean live, boolean ready, boolean inMaintanance, boolean readyInMainenance)
+            throws Exception {
+        String content = String.format("%s=%s\n%s=%s\n%s=%s\n%s=%s",
+                FailSafeDummyService.LIVE_CONFIG_PROPERTY, live,
+                FailSafeDummyService.READY_CONFIG_PROPERTY, ready,
+                FailSafeDummyService.READY_IN_MAINTENANCE_CONFIG_PROPERTY, readyInMainenance,
+                FailSafeDummyService.IN_MAINTENANCE_CONFIG_PROPERTY, inMaintanance);
+        // TODO Java 11 API way - Files.writeString(propertyFilePath, INCREMENT_CONFIG_PROPERTY + "=" + increment);
+        Files.write(propertyFile, content.getBytes(StandardCharsets.UTF_8));
+    }
+
+    /**
+     * Setup a microprofile-config-smallrye subsystem to obtain values from {@link CustomConfigSource}
+     */
+    static class SetupTask implements ServerSetupTask {
+        private static final String TEST_MODULE_NAME = "test.custom-config-source";
+
+        @Override
+        public void setup(ManagementClient managementClient, String s) throws Exception {
+            OnlineManagementClient client = ManagementClientProvider.onlineStandalone(managementClient);
```

Comment:
Please close client.

Reply:
Actually, the client can not be closed. Doing so would also close `managementClient` managed by arquillian which is something we do not want to. I would not be able to deploy then since  `managementClient` would be closed.

Reply:
Thanks - I understand limitations which you are running into but there is still an open client in the end...

Reply:
I was looking at the code of Arquillian extension and it looks like that `ManagementClient` is simply injected there as in ordinary tests where we don't close it as well. However I'm not sure what is the object lifecycle for `@Inject`. Still I don't see as issue.